### PR TITLE
Add basic rust integration test. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -721,6 +729,21 @@ jobs:
             core0.test_hello_argc
             core2.test_demangle_stacks_symbol_map"
       - upload-test-results
+  test-rust:
+    executor: linux-python
+    #docker:
+    #  - image: cimg/rust:1.82.0-node
```
Comment (tlively) [inline .circleci/config.yml:735]: Stale?

File: .circleci/config.yml [hunk 2]
```diff
@@ -956,59 +982,4 @@ jobs:
 workflows:
   build-test:
     jobs:
-      - flake8
```
Comment (tlively) [inline .circleci/config.yml:959]: Why are all these jobs deleted? Was that just for testing this change?

Comment (sbc100) [reply .circleci/config.yml:959]: Sorry, just for testing.

**Comment by sbc100** [comment]
Sorry I'm still trying to get the CI working for this.

**Comment by sbc100** [comment]
OK, all looks good now, PTAL

File: .circleci/config.yml [hunk 3]
```diff
@@ -70,6 +70,15 @@ commands:
       - run:
           name: pip install
           command: << parameters.python >> -m pip install -r requirements-dev.txt
+  install-rust:
+    steps:
+      - run:
+          name: install rust
+          command: |
+            curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
```
Comment (dschuff) [inline .circleci/config.yml:78]: #YOLO

File: .circleci/config.yml [hunk 4]
```diff
@@ -721,6 +730,18 @@ jobs:
             core0.test_hello_argc
             core2.test_demangle_stacks_symbol_map"
       - upload-test-results
+  test-rust:
+    executor: linux-python
```
Comment (dschuff) [inline .circleci/config.yml:734]: Another option here might be the CircleCI rust image? https://circleci.com/developer/images/image/cimg/rust

Comment (sbc100) [reply .circleci/config.yml:734]: I tried that first but decided against it for 2 reasons:

1. We need python to run emscripten and the rust image does not have python installed.
2. We want to run rust nightly I think, which the rust image doesn't seem to have

File: .circleci/config.yml (hunk 3 above)
Comment (sbc100) [reply .circleci/config.yml:78]: This is the official way to install rust apparently.. so its actually good that we use this method here I think.

**Comment by sbc100** [comment]
CC @hoodmane in case you have ideas for some more rust smoke tests we can/should add here?

File: .circleci/config.yml (hunk 4 above)
Comment (dschuff) [reply .circleci/config.yml:734]: Ah, interesting. Are there changes going into core rustc that we will need?

Comment (sbc100) [reply .circleci/config.yml:734]: As an aside I find it odd that circle CI doesn't have base images with more than one tool.  Surely there are a lot of polyglot projects out there that needs python + node or python + rust.   It seems that such projects currently have to build their own base image or manually install one or more runtimes.

**Review by tlively — APPROVED** [review]

**Comment by kripken** [comment]
@sbc100 Rust tests are failing on the roller:

https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8730720020817447969/+/u/Emscripten_testsuite__other_/stdout

https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8730716182753081457/+/u/Emscripten_testsuite__other_/stdout

**Comment by dschuff** [comment]
That should be fixed by https://chromium-review.googlesource.com/c/emscripten-releases/+/6038095

File: .circleci/config.yml (hunk 4 above)
Comment (sbc100) [reply .circleci/config.yml]: What's the recommended way to do that?

Comment (curiousdannii) [reply .circleci/config.yml]: `rustup toolchain install nightly`

Looks like you might also be able to specify it when you install rustup https://rust-lang.github.io/rustup/installation/index.html#installing-nightly

**Comment by kripken** [comment]
Hmm, that was merged 2PM yesterday, but the roller is still failing as of 4 hours ago. I see that the error message

https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8730651596731910801/+/u/Emscripten_testsuite__other_/stdout

mentions `EMTEST_SKIP_CARGO` while the code here mentions `_RUST` instead. Are those perhaps not in sync?

**Comment by sbc100** [comment]
Ah yes!  I decided to fix this on the emscripten side: https://github.com/emscripten-core/emscripten/pull/22982

**Pull request merged**